src/client/dfs/dfs_sys.c

```diff
@@ -227,8 +227,7 @@ hash_lookup(dfs_sys_t *dfs_sys, struct sys_path *sys_path)
 	rc = dfs_lookup(dfs_sys->dfs, sys_path->dir_name, O_RDWR, &hdl->obj,
 			&mode, NULL);
 	if (rc != 0) {
-		D_ERROR("dfs_lookup() %s failed: %s\n",
-			sys_path->dir_name, strerror(rc));
+		D_DEBUG(DB_TRACE, "dfs_lookup() %s failed: %s\n", sys_path->dir_name, strerror(rc));
```

Comment:
I'm okay with this (debug probably makes more sense anyway), but how does this related to fault injection? (genuine question)

Reply:
@daltonbohning for some reason it was giving a "warning in strict mode" in the nlt-errors.json file. I am not sure why, but using DEBUG instead removed the error from the file.

Reply:
If it's an error then use D_ERROR().

The actual code doing the log check is at https://github.com/daos-stack/daos/blob/master/src/tests/ftest/cart/util/cart_logtest.py#L401-L424 and it's attempting to verify that you're using the DF_RC macro but that only applies to daos error numbers, for system error numbers it's doing a much looser check that the error number itself (12) is printed.  I think in this case changing back the line and modifying cart_lotest.py to accept the strerror() string.

Reply:
@ashleypittman This was the only D_ERROR in this file. Everything else is DEBUG. The idea was that most of these functions are pass-through / shorthand for DFS. So, either (1) DFS itself would log appropriate error messages, or (2) the caller of the dfs_sys functions would check the return codes.